enterprise/frontend/src/embedding-sdk/components/public/DashboardCreateModal/use-dashboard-create.unit.spec.tsx

```diff
@@ -0,0 +1,94 @@
+import userEvent from "@testing-library/user-event";
+import fetchMock from "fetch-mock";
+
+import { setupDashboardCreateEndpoint } from "__support__/server-mocks";
+import { screen, renderWithProviders } from "__support__/ui";
+import { createMockJwtConfig } from "embedding-sdk/test/mocks/config";
+import type { Dashboard } from "metabase-types/api";
+
+import type { DashboardCreateParameters } from "./use-dashboard-create";
+import { useDashboardCreate } from "./use-dashboard-create";
+
+const TEST_COLLECTION_ID = 11;
+const TEST_DASHBOARD_ID = 123;
+const TEST_DASHBOARD_NAME = "TestDashboard";
+
+describe("useDashboardCreate", () => {
+  it('should create a new dashboard after "createDashboard" is called', async () => {
+    setup();
+
+    await userEvent.click(screen.getByText("Create dashboard"));
+
+    expect(
+      fetchMock.calls(`path:/api/dashboard`, { method: "POST" }),
+    ).toHaveLength(1);
+  });
+
+  it("should return dashboard api call response", async () => {
+    const { onDashboardCreateSpy } = setup();
+
+    await userEvent.click(screen.getByText("Create dashboard"));
+
+    expect(onDashboardCreateSpy).toHaveBeenCalledTimes(1);
+    expect(onDashboardCreateSpy).toHaveBeenLastCalledWith(
+      expect.objectContaining({
+        id: TEST_DASHBOARD_ID,
+        name: TEST_DASHBOARD_NAME,
+      }),
+    );
+  });
+});
+
+const TestComponent = (
+  props: DashboardCreateParameters & {
+    onDashboardCreate: (dashboard: Dashboard) => void;
+  },
+) => {
+  const { onDashboardCreate, ...restProps } = props;
+  const { createDashboard } = useDashboardCreate();
+
+  const handleCreate = async () => {
+    const dashboard = await createDashboard(restProps);
+
+    onDashboardCreate(dashboard);
+  };
+
+  return (
+    <div>
+      <button onClick={handleCreate}>Create dashboard</button>
+    </div>
+  );
+};
+
+function setup(
+  { props }: { props: DashboardCreateParameters } = {
```

Comment:
since we never pass anything to `setup` I'd just declare this object inside the setup itself to make it clear.

you can even combine `TEST_DASHBOARD_ID` and `TEST_DASHBOARD_NAME` into a single object.